src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/OffenderSearchService.kt

```diff
@@ -0,0 +1,28 @@
+package uk.gov.justice.digital.hmpps.incentivesapi.service
+
+import org.springframework.stereotype.Service
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.awaitBody
+import uk.gov.justice.digital.hmpps.incentivesapi.dto.OffenderSearchPrisonerResponse
+
+@Service
+class OffenderSearchService(private val offenderSearchWebClient: WebClient) {
+  /**
+   * Searches for offenders using a cell location prefix, e.g. MDI-1
+   * Requires role ROLE_PRISONER_IN_PRISON_SEARCH or ROLE_PRISONER_SEARCH
```

Comment:
I think we will need to add these roles to the `hmpps-incentives-ui` client credentials

Reply:
indeed! i just wanted it documented _somewhere_!

Reply:
Thanks. I've added `ROLE_PRISONER_SEARCH` to `hmpps-incentives-ui` client creds in dev. Once we've validated it works I'll update preprod and prod too

Reply:
I'm not saying to change but I do wonder if these roles could or should be added to the resource via `@PreAuthorize("hasAnyRole('ROLE_PRISONER_IN_PRISON_SEARCH', 'ROLE_PRISONER_SEARCH')")`

Maybe not, I guess these are permissions for the upstream API but in practice the client would need these and maybe that would be more explicit to check upfront? More of an open question 🤔

Reply:
yeah i think the permissions ought to be managed upstream and i would prefer not to replicate the checks in case upstream changes role requirements